File: functionapi-api/src/main/java/io/github/cottonmc/functionapi/api/content/ItemTemplate.java
package io.github.cottonmc.functionapi.api.content;


import io.github.cottonmc.functionapi.api.content.enums.CreativeTabs;
import io.github.cottonmc.functionapi.api.content.enums.ItemType;
import io.github.cottonmc.functionapi.api.script.FunctionAPIIdentifier;

/**
 * Stores enough data to create a new item.
* */
public interface ItemTemplate {


    int maxCount();
    ItemType getType();
    void setType(ItemType type);
    boolean isColored();
    void setLootId(String id);
    String getLootid();
    void setMaxCount(int amount);

    int getMaxUseTime();
    void setMaxUseTime(int maxUseTime);

    void setTab(CreativeTabs creativeTabs);
    void setTab(String creativeTabs, FunctionAPIIdentifier item);
    boolean isDefaultTab();

    CreativeTabs getDefaultTab();
    String getCustomTabName();
    FunctionAPIIdentifier getCreativeTabItem();
}

File: functionapi-api/src/main/kotlin/io/github/cottonmc/functionapi/api/content/ItemTemplate.kt
package io.github.cottonmc.functionapi.api.content

import io.github.cottonmc.functionapi.api.FunctionAPIIdentifier
import io.github.cottonmc.functionapi.api.content.enums.CreativeTabs
import io.github.cottonmc.functionapi.api.content.enums.ItemType
import io.github.cottonmc.functionapi.util.datacommand.ArgumentSetter
import io.github.cottonmc.functionapi.util.datacommand.Context
import io.github.cottonmc.functionapi.util.datacommand.Name

/**
 * Stores enough data to create a new item.
 */
@Name("itemsettings")
@Context("item")
interface ItemTemplate {
    var maxCount: Int
        @Name("stack_size", "amount") @ArgumentSetter set
    var type: ItemType
        @Name("type", "type") @ArgumentSetter set
    var maxUseTime: Int
        @Name("max_use_time", "use time") @ArgumentSetter set
    var creativeTab: CreativeTabs
        @Name("creative_tab", "tab") @ArgumentSetter set
}